Exiled.Events/Patches/Events/Scp049/StartingAndFinishingRecall.cs

```diff
@@ -7,181 +7,124 @@
+
+                // var ev = new StartingRecallEventArgs(...)
+                new CodeInstruction(OpCodes.Newobj, GetDeclaredConstructors(typeof(StartingRecallEventArgs))[0]),
+                new CodeInstruction(OpCodes.Dup),
+
+                // Handlers.Scp049.OnStartingRecall(ev)
+                new CodeInstruction(OpCodes.Call, Method(typeof(Handlers.Scp049), nameof(Handlers.Scp049.OnStartingRecall))),
+
+                // if (!ev.IsAllowed)
+                //   return;
+                new CodeInstruction(OpCodes.Callvirt, PropertyGetter(typeof(StartingRecallEventArgs), nameof(StartingRecallEventArgs.IsAllowed))),
+                new CodeInstruction(OpCodes.Brfalse_S, returnLabel),
+            });
+
+            offset = -5;
+
+            // Search for the first "RoundSummary.changed_into_zombies++", then subtract the offset to get "ldstr".
+            index = newInstructions.FindIndex(instruction => instruction.opcode == OpCodes.Ldsfld &&
+            (FieldInfo)instruction.operand == Field(typeof(RoundSummary), nameof(RoundSummary.changed_into_zombies))) + offset;
+
+            // var ev = new FinishingRecallEventArgs(Player.Get(referenceHub), Player.Get(this.Hub), true);
+            //
+            // Handlers.Scp049.OnFinishingRecall(ev)
+            //
+            // if (!ev.IsAllowed)
+            //   return;
+            newInstructions.InsertRange(index, new[]
             {
-                Exiled.API.Features.Log.Error($"Exiled.Events.Patches.Events.Scp049.StartingAndFinishingRecall: {e}\n{e.StackTrace}");
+                // Player.Get(referenceHub)
+                new CodeInstruction(OpCodes.Ldloc_S, 8).MoveLabelsFrom(newInstructions[index]),
+                new CodeInstruction(OpCodes.Call, Method(typeof(Player), nameof(Player.Get), new[] { typeof(ReferenceHub) })),
+
+                // Player.Get(this.Hub)
+                new CodeInstruction(OpCodes.Ldarg_0),
+                new CodeInstruction(OpCodes.Ldfld, Field(typeof(Scp049), nameof(Scp049.Hub))),
+                new CodeInstruction(OpCodes.Call, Method(typeof(Player), nameof(Player.Get), new[] { typeof(ReferenceHub) })),
+
+                // true
+                new CodeInstruction(OpCodes.Ldc_I4_1),
+
+                // var ev = new FinishingRecallEventArgs(...)
+                new CodeInstruction(OpCodes.Newobj, GetDeclaredConstructors(typeof(FinishingRecallEventArgs))[0]),
+                new CodeInstruction(OpCodes.Dup),
+
+                // Handlers.Scp049.OnFinishingRecall(ev)
+                new CodeInstruction(OpCodes.Call, Method(typeof(Handlers.Scp049), nameof(Handlers.Scp049.OnFinishingRecall))),
+
+                // if (!ev.IsAllowed)
+                //   return;
+                new CodeInstruction(OpCodes.Callvirt, PropertyGetter(typeof(FinishingRecallEventArgs), nameof(FinishingRecallEventArgs.IsAllowed))),
+                new CodeInstruction(OpCodes.Brfalse_S, returnLabel),
```

Comment:
Use `brfalse` instead

Reply:
Ok, since it's a long method, it could be useful to not make it break if they add more instructions in the future.